invokeai/app/invocations/latent.py

```diff
@@ -655,7 +813,15 @@ def init_scheduler(
         timesteps = timesteps[t_start_idx : t_start_idx + t_end_idx]
         num_inference_steps = len(timesteps) // scheduler.order
 
-        return num_inference_steps, timesteps, init_timestep
+        scheduler_step_kwargs = {}
+        scheduler_step_signature = inspect.signature(scheduler.step)
+        if "generator" in scheduler_step_signature.parameters:
+            # At some point, someone decided that schedulers that accept a generator should use the original seed with
+            # all bits flipped. I don't know the original rationale for this, but now we must keep it like this for
+            # reproducibility.
+            scheduler_step_kwargs = {"generator": torch.Generator(device=device).manual_seed(seed ^ 0xFFFFFFFF)}
```

Comment:
They needed an explicitly seeded generator passed for reproducibility.

Nobody wanted to add a new `stochastic_scheduler_seed` or `generator` field to DenoiseLatentsInvocation, but they figured out how to get a seed from one of its input LatentFields.

However, I feared re-using the same `seed` as had already been used to generate that input would result in the same tensor, which is not the random behavior those schedulers _expect_ to be doing.

So the compromise was, when you have an operation that needs a seed but the only seed defined has already been used earlier in the graph, you do some reproducible transformation like this to get a new seed for your new operation.

FWIW, I recommend factoring this `seed  ^ 0xFFFFFFFF` up out of this method to the place that calls `get_scheduler` and `init_scheduler`. That makes this method simpler: the seed passed to `init_scheduler` is, in fact, the seed used for them; and the quirky logic about deriving the `scheduler_seed` goes with the code that's trying to pull a `seed` out of the InvocationContext's LatentFields.